nemo/lightning/run/plugins.py

```diff
@@ -0,0 +1,121 @@
+import copy
+import logging
+import os
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Callable, Optional
+
+import nemo_run as run
+import yaml
+from nemo_run.core.serialization.yaml import YamlSerializer
+from pytorch_lightning import Callback
+from pytorch_lightning.loggers import WandbLogger
+
+from nemo.lightning.pytorch.callbacks import NsysCallback, PreemptionCallback
+
```

Comment:
Can we add documentation explaining the difference between plugins and callbacks? And about when to setup one or the other?

Reply:
Yeah I think we can add this in the documentation site. Wdyt?

Reply:
Added in https://github.com/NVIDIA/NeMo/pull/10223/commits/eb2630380326a48497c40b8e390bd4bab793ef83